### ISSUE TITLE
Table instances created with the same arguments do NOT evaluate as equals

### ISSUE THREAD

**Issue by akettmann-e24** [googleapis/python-bigquery#854]
Two table objects with the same arguments do not evaluate as equal. This makes unit testing and comparing calls to `Client.insert_rows` more complicated than necessary.

#### Environment details

  - OS type and version: Fedora 32
  - Python version: `3.9.5`
  - pip version: `21.1.3`
  - `google-cloud-bigquery` version: `2.23.2`

#### Steps to reproduce

  1. Create table with set of args
  2. Create separate instance of table with same set of args
  3. Test equality `t1 == t2`
  4. returns false

#### Code example

```python
from google.cloud.bigquery import Table
t1 = Table('project.dataset.table')
t2 = Table('project.dataset.table')
t1 == t2
# returns: False
```

**Comment by plamut** [comment]
That sounds reasonable, we could indeed implement custom comparison logic for `Table` classes.

**Comment by akettmann-e24** [comment]
Makes sense to me! Wasn't sure which way I should put it in, but I figured if I am hitting it I am likely not the only one.
